packages/team-management/src/components/TeamsAddEdit/Form.tsx

```diff
@@ -0,0 +1,257 @@
+  values: FormField,
+  initialValue: FormField,
+  id?: string
+) {
+  setIsSubmitting(true);
+  const Teamid = id ?? v4();
+
+  const payload: OrganizationPOST = {
+    active: values.active,
+    identifier: Teamid,
+    name: values.name,
+    type: {
+      coding: [
+        {
+          code: 'team',
+          display: 'Team',
+          system: 'http://terminology.hl7.org/CodeSystem/organization-type',
+        },
+      ],
+    },
+  };
+
+  setTeam(accessToken, payload, id)
+    .then(async () => {
+      // Filter and seperate the practitioners uuid
+      // const toBe = initialValue.practitioners.filter((val) => values.practitioners.includes(val));
+      const toAdd = values.practitioners.filter((val) => !initialValue.practitioners.includes(val));
+      const toRem = initialValue.practitioners
+        .filter((val) => !values.practitioners.includes(val))
+        .filter((e) => e !== '');
+
+      await SetPractitioners(practitioner, toAdd, toRem, accessToken, Teamid);
+      history.goBack();
+    })
+    .catch(() => sendErrorNotification('An error occurred'))
+    .finally(() => setIsSubmitting(false));
+}
+
+/**
+ * handle Practitioners
+ *
+ * @param {Practitioner} practitioner list of practitioner to filter the selected one from
+ * @param {Array<string>} toAdd list of practitioner uuid to add
+ * @param {Array<string>} toRemove list of practitioner uuid to remove
+ * @param {string} accessToken Token for api calles
+ * @param {string} id id of the team
+ */
+async function SetPractitioners(
+  practitioner: Practitioner[],
+  toAdd: string[],
+  toRemove: string[],
+  accessToken: string,
+  id: string
+) {
+  sendInfoNotification('Assigning Practitioners');
+
+  // Api Call to delete practitioners
+  toRemove.forEach((prac) => {
+    const serve = new OpenSRPService(accessToken, API_BASE_URL, PRACTITIONER_DEL + prac);
+    serve.delete().catch(() => sendErrorNotification('An error occurred'));
+  });
+
+  // Api Call to add practitioners
+  const toAddPractitioner = practitioner.filter((e) => toAdd.includes(e.identifier));
+  const payload: PractitionerPOST[] = toAddPractitioner.map((prac) => {
+    return {
+      active: prac.active,
+      identifier: v4(),
+      practitioner: prac.identifier,
+      organization: id,
+      code: { text: 'Community Health Worker' },
+    };
+  });
+
+  const serve = new OpenSRPService(accessToken, API_BASE_URL, PRACTITIONER_POST);
+  await serve.create(payload);
+
+  sendSuccessNotification('Successfully Assigned Practitioners');
+}
+
+/**
+ * Function to make teams API call
+ *
+ * @param {string} accessToken Token for api calles
+ * @param {OrganizationPOST} payload payload To send
+ * @param {string} id of the team if already created
+ */
+export async function setTeam(accessToken: string, payload: OrganizationPOST, id?: string) {
+  if (id) {
+    const serve = new OpenSRPService(accessToken, API_BASE_URL, TEAMS_PUT + id);
+    await serve.update(payload);
```

Comment:
can we add a catch block to handle errors that might be thrown by API

Reply:
@ashfahan

Reply:
@KipSigei can you explain a bit more like what kind of error could it be ❓

Reply:
add something similar to this https://github.com/OpenSRP/web/pull/195/files#diff-48718deb9f4a0318a99a36fd28e223e0b65cce66273134d0a6c854c17da5c8f3R64

Reply:
done